mods/default/functions.lua

```diff
@@ -715,6 +715,37 @@ function default.register_craft_metadata_copy(ingredient, result)
 	end)
 end
 
+--
+-- Log API / helpers
+--
+
+local log_non_player_actions = minetest.settings:get_bool("log_non_player_actions") ~= false
+
+function default.log_player_action(player, message, pos)
+	local who = player:get_player_name()
+	if not player.is_fake_player and player:is_player() then
+		-- log action of real player
+		minetest.log("action",  who .. " " .. message .. " at " .. minetest.pos_to_string(pos))
+	elseif log_non_player_actions then
+		-- log action of non real player
+		who = who .. "(" .. (type(player.is_fake_player) == "string"
+			and player.is_fake_player or "*").. ")"
+		minetest.log("action",  who .. " " .. message .. " at " .. minetest.pos_to_string(pos))
```

Comment:
You may move the call to `minetest.log` to the bottom of the function to deduplicate it.